rholang/src/test/scala/rholang/rosette/CompilerTests.scala

```diff
@@ -20,11 +20,11 @@ class CompilerTests extends FunSuite {
 
   def runtime = Runtime.create(Files.createTempDirectory(tmpPrefix), mapSize)
 
-  val testFiles: Iterable[Path] =
-    Files.newDirectoryStream(Paths.get("tests")).asScala
+  val testFiles: Iterator[Path] =
+    Files.walk(Paths.get("tests")).iterator().asScala
 
-  val failureTestFiles: Iterable[Path] =
-    Files.newDirectoryStream(Paths.get("failure_tests")).asScala
+  val failureTestFiles: Iterator[Path] =
+    Files.walk(Paths.get("failure_tests")).iterator().asScala
```

Comment:
What was wrong with previous implementation?

Reply:
It wasn't traversing the `tests` directory recursively. So rholang files under `tests/mercury-tut` weren't even tested before and I assume they should.

Reply:
OK, nice catch :+1: